### PR TITLE
Issue #SB-16574: Rewrite the tpd reports with analytics data products…

### PR DESCRIPTION
**Pull request by utk14**
… framework

### CODE REVIEW

File: batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseConsumptionModel.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      val configMap = config("reportConfig").asInstanceOf[Map[String, AnyRef]]
+      val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(configMap))
+
+      import sqlContext.implicits._
+      reportConfig.output.map { f =>
+        if (f.`type`.equals("csv")) {
+          val df = data.toDF().na.fill(0L)
+          val storageService = fc.getStorageService("azure")
+          CourseUtils.postDataToBlob(df, f,config, storageService)
+          storageService.closeContext()
```
Comment (SanthoshVasabhaktula) [inline batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseConsumptionModel.scala]: 1. Remove the storage service implementation
2. Do not hardcode it for csv format

File: batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseEnrollmentModel.scala [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      val configMap = config("reportConfig").asInstanceOf[Map[String, AnyRef]]
+      val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(configMap))
+
+      import sqlContext.implicits._
+      reportConfig.output.map { f =>
+        if (f.`type`.equals("csv")) {
+          val df = data.toDF().na.fill(0L)
+          val storageService = fc.getStorageService("azure")
+          CourseUtils.postDataToBlob(df, f,config, storageService)
+          storageService.closeContext()
```
Comment (SanthoshVasabhaktula) [inline batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseEnrollmentModel.scala]: 1. Remove the storage service implementation
2. Do not hardcode it for csv format

File: batch-models/src/main/scala/org/ekstep/analytics/util/CourseUtils.scala [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        col("status").as("status")
+      )
+  }
+
+  def getTenantInfo()(implicit sc: SparkContext, sqlContext: SQLContext): DataFrame = {
+    val sunbirdKeyspace = AppConf.getConfig("course.metrics.cassandra.sunbirdKeyspace")
+    loadData(Map("table" -> "organisation", "keyspace" -> sunbirdKeyspace)).select("slug","id")
+  }
+
+  def postDataToBlob(data: DataFrame, outputConfig: OutputConfig, config: Map[String, AnyRef], storageService: BaseStorageService)(implicit sc: SparkContext, fc: FrameworkContext) = {
```
Comment (SanthoshVasabhaktula) [inline batch-models/src/main/scala/org/ekstep/analytics/util/CourseUtils.scala]: StorageService is no longer required

File: batch-models/src/main/scala/org/ekstep/analytics/util/CourseUtils.scala [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    val renamedDf = filteredDf.select(filteredDf.columns.map(c => filteredDf.col(c).as(labelsLookup.getOrElse(c, c))): _*)
+    val reportFinalId = if (outputConfig.label.nonEmpty && outputConfig.label.get.nonEmpty) reportConfig.id + "/" + outputConfig.label.get else reportConfig.id
+
+    val finalDf = renamedDf.na.replace("status",Map("0"->BatchStatus(0).toString, "1"->BatchStatus(1).toString, "2"->BatchStatus(2).toString))
+    finalDf.show()
+    saveReport(data, config ++ Map("dims" -> dimsLabels, "reportId" -> reportFinalId, "fileParameters" -> outputConfig.fileParameters))
+  }
+
+  def saveReport(data: DataFrame, config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): Unit = {
+    CommonUtil.setStorageConf(config.getOrElse("store", "local").toString, config.get("accountKey").asInstanceOf[Option[String]], config.get("accountSecret").asInstanceOf[Option[String]])
```
Comment (SanthoshVasabhaktula) [inline batch-models/src/main/scala/org/ekstep/analytics/util/CourseUtils.scala]: Move this to pre-process.

File: batch-models/src/main/scala/org/ekstep/analytics/util/FileUtil.scala [hunk 5]
```diff
@@ -7,6 +7,8 @@ import org.ekstep.analytics.framework.Level.INFO
 import org.ekstep.analytics.framework.util.JobLogger
 import org.sunbird.cloud.storage.conf.AppConf
 import org.sunbird.cloud.storage.factory.{StorageConfig, StorageServiceFactory}
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.SparkContext
```
Comment (SanthoshVasabhaktula) [inline batch-models/src/main/scala/org/ekstep/analytics/util/FileUtil.scala]: Why did this file change? Can you revert the changes?

Comment (utk14) [reply batch-models/src/main/scala/org/ekstep/analytics/util/FileUtil.scala]: Done

File: batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseEnrollmentModel.scala (hunk 2 above)
Comment (utk14) [reply batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseEnrollmentModel.scala]: Done

File: batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseConsumptionModel.scala (hunk 1 above)
Comment (utk14) [reply batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseConsumptionModel.scala]: Done

File: batch-models/src/main/scala/org/ekstep/analytics/util/CourseUtils.scala (hunk 4 above)
Comment (utk14) [reply batch-models/src/main/scala/org/ekstep/analytics/util/CourseUtils.scala]: Done
